### PR TITLE
:seedling: Add a fuzzer function for *metav1.Time

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

Currently, when metav1.Time is a pointer it never gets fuzzed by the
meta functions. After a further look, it seems like the Creation or
Deletion timestamps on ObjectMeta are fuzzed in this way as well.

This patch adds custom logic to fuzz the metav1.Time objects when it's
not nil.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

/milestone v0.4
/assign @sbueringer

### CODE REVIEW

File: util/conversion/conversion.go [hunk 1]
```diff
@@ -128,7 +129,26 @@ func UnmarshalData(from metav1.Object, to interface{}) (bool, error) {
...
+		metafuzzer.Funcs,
+		func(_ runtimeserializer.CodecFactory) []interface{} {
+			return []interface{}{
+				// Custom fuzzer for metav1.Time pointers which weren't
+				// fuzzed and always resulted in `nil` values.
+				// This implementation is somewhat similar to the one provided
+				// in the metafuzzer.Funcs.
+				func(input *metav1.Time, c fuzz.Continue) {
+					if input != nil {
+						var sec, nsec uint32
```
Comment (sbueringer) [inline util/conversion/conversion.go:142]: Is there a reson to not use `int64` directly here?

Like in:
```go
			var sec, nsec int64
			c.Fuzz(&sec)
			c.Fuzz(&nsec)
			j.CreationTimestamp = metav1.Unix(sec, nsec).Rfc3339Copy()
```

**Comment by sbueringer** [comment]
Related to https://github.com/kubernetes-sigs/cluster-api/pull/4535#discussion_r622438922

File: util/conversion/conversion.go (hunk 1 above)
Comment (vincepri) [reply util/conversion/conversion.go:142]: int64, being large and allowing negative numbers generates invalid years when performing the rfc339 copy

**Comment by vincepri** [comment]
Both are ok

File: util/conversion/conversion.go (hunk 1 above)
Comment (sbueringer) [reply util/conversion/conversion.go:142]: Makes sense.

**Pull request merged**